app/models/provider.rb

```diff
@@ -37,6 +37,10 @@ def from_previous_recruitment_cycle
       .first
   end
 
+  def declared_visa_sponsorship?
+    %i[can_sponsor_student_visa can_sponsor_skilled_worker_visa].none? { |attr| send(attr).nil? }
```

Comment:
Minor code style suggestion (took me a min to parse the "none are nil" format)

Shame there's no `not_nil?` method in ruby!

```suggestion
    !can_sponsor_student_visa.nil? && !can_sponsor_skilled_worker_visa.nil?
```